src/pages/Admin/AdminCohortsPage/AddNewCohortModal.tsx

```diff
@@ -0,0 +1,115 @@
+import React, { useState } from 'react';
+import styled from 'styled-components';
+import { useForm } from 'react-hook-form';
+
+import { Input, InputField } from '../../../components/Input';
+import { Modal } from '../../../components/Modal/Modal';
+import { Button, PrimaryButton } from '../../../components/Button';
+import {
+  adminCreateCohort,
+  AdminCreateCohortProps,
+} from '../../../api/adminCreateCohort';
+import { getAdminCohorts } from '../../../api/getAdminCohorts';
+import { Cohort, ErrorMessageType } from '../../../types';
+import {
+  FormErrorMessages,
+  StyledErrorMessage,
+} from '../../../components/ErrorMessage';
+import { cohortValidationRules } from '../../../utilities/validation';
+
+export interface AddNewCohortModalProps {
+  isOpen: boolean;
+  toggle(): void;
+  setCohortsList(value: Cohort[]): void;
+}
+
+const defaultCohortValues = {
+  name: '',
+  startDate: '',
+  endDate: '',
+};
+
+const StyledForm = styled.form`
+  display: flex;
+  flex-direction: column;
+  gap: 1em;
+`;
+
+export const AddNewCohortModal = ({
+  isOpen,
+  toggle,
+  setCohortsList,
+}: AddNewCohortModalProps) => {
+  const [errorMessages, setErrorMessages] = useState<ErrorMessageType[]>([]);
+
+  const {
+    register,
+    handleSubmit,
+    reset,
+    formState: { errors: formErrors },
+  } = useForm<AdminCreateCohortProps>();
+
+  const handleCloseModal = () => {
+    toggle();
+    reset(defaultCohortValues);
+  };
+
+  const onSubmit = async (payload: AdminCreateCohortProps) => {
+    try {
+      await adminCreateCohort(payload);
+      const { data } = await getAdminCohorts();
```

Comment:
I will leave this as is for now, I tried updating the state, it works but if the user navigates away from the page from the navbar, and then come back to this page, it'll lose the local state. So I think I'll have to redo the authorized pages part, might as well add the react cookie with it.